src/styles/createPalette.js

```diff
@@ -153,13 +137,20 @@ export default function createPalette(palette: Object) {
     return contrastText;
   }
 
-  if (!primary.contrastText) {
-    primary.contrastText = getContrastText(primary.main);
+  function augmentColor(color, mainShade, lightShade, darkShade) {
+    if (!color.main && color[mainShade]) {
+      color.main = color[mainShade];
+    }
+    addLightOrDark(color, 'light', lightShade, tonalOffset);
+    addLightOrDark(color, 'dark', darkShade, tonalOffset);
+    if (!color.contrastText) {
+      color.contrastText = getContrastText(color.main);
+    }
   }
 
-  if (!secondary.contrastText) {
-    secondary.contrastText = getContrastText(secondary.main);
-  }
+  augmentColor(primary, 500, 300, 700);
+  augmentColor(secondary, 'A400', 'A200', 'A700');
+  augmentColor(error, 500, 300, 700);
```

Comment:
Are these the appropriate shades to use for error augmentation?

Reply:
@pelotom I missed this because the PR was tagged typescript, but why add `error.light`, `error.dark` and `error.contrastText` to the theme when they aren't needed?

Reply:
https://github.com/mui-org/material-ui/pull/9973#issuecomment-359209304

Reply:
Sorry, didn't see the earlier comments as I'd gone straight to the code.

I don't entirely agree. They're now the only keys in the theme that aren't actually used for something, and someone customizing it (as unlikely as it is to change the error color) might think they need to provide colors for those keys. 

NVM, It's done now.

Reply:
No reason we couldn't revert it. I don't feel strongly one way or another, it just struck me as inconsistent with the treatment of `primary` and `secondary`, and I imagined users of the palette might find it surprising that they couldn't access `light` and `dark` variants on `error`.

Reply:
We need to update the docs in that case: https://material-ui-next.com/customization/themes/